src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -17,14 +17,14 @@ import {
   toBN,
   toBNWei,
 } from "../utils";
+import { Transport } from "viem";
 
 // This needs to be implemented for every chain and passed into RelayFeeCalculator
 export interface QueryInterface {
   getGasCosts: (
     deposit: Deposit,
     relayer: string,
-    gasPrice?: BigNumberish,
-    gasLimit?: BigNumberish
+    options?: Partial<{ gasPrice: BigNumberish; gasUnits: BigNumberish; omitMarkup: boolean; transport: Transport }>
```

Comment:
I think it'd be nice to avoid having `Transport` bleed out into the `QueryInterface` definitions. I'm not sure if that's acheivable atm and it may not be actionable right now. But in general I think it would be good to keep the implementation choices of the gas price oracle defined internally - especially since it's only really used for test (...it might hint at the need to define a special class for test? idk).

Reply:
So currently we use `getGasCosts` directly in the API. That's why I had to expose it here 🤔

Reply:
But yea agree it would be nicer to keep it internal